runtime/src/attestation/tests.rs

```diff
@@ -18,8 +18,7 @@
 
 use super::*;
 use parity_codec::Encode;
-use primitives::*;
-use primitives::{ed25519 as x25519, Blake2Hasher, H256};
+use primitives::{ed25519 as x25519, Blake2Hasher, H256, *};
```

Comment:
Either we import everything by name or we import using the wildcard. We could remove `Blake2Hasher, H256`. But I don't like wildcard imports. There might even be an option to convert wildcard imports to explicit imports in RLS.

Reply:
ok let's see what we can do about these

Reply:
```
use primitives::{ed25519, ed25519 as x25519, Blake2Hasher, H256, Pair};
```
works here - but we'd have to suppress unused imports warnings

Reply:
And we need the Pair import?

Reply:
Ah yes you can remove `ed25519` from the imports. There should be no unused warning after that...

Reply:
we need the pair import: 
```
error[E0599]: no function or associated item named `from_seed` found for struct `substrate_primitives::ed25519::Pair` in the current scope
   --> runtime/src/attestation/tests.rs:246:36
    |
246 |         let pair_charlie = x25519::Pair::from_seed(*b"Charlie                         ");
    |                                          ^^^^^^^^^ function or associated item not found in `substrate_primitives::ed25519::Pair`
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use substrate_primitives::crypto::Pair;
```
and the Pair import then required the ed25519 import

Reply:
`use primitives::{ed25519 as x25519, Blake2Hasher, H256, Pair};` works for me.

Reply:
ah ok. I see. You get a clippy warning there. That's strange and probably a bug in clippy...

Reply:
Try calling clippy with `cargo clippy --tests`. It's probably unused because clippy doesn't see the test methods or something like that...